plaso/parsers/winreg.py

```diff
@@ -169,23 +196,25 @@ def _ParseRecurseKeys(self, parser_mediator, root_key):
       if parser_mediator.abort:
         break
 
-      matching_plugin = None
+      self._ParseKey(parser_mediator, registry_key)
 
-      normalized_key_path = self._NormalizeKeyPath(registry_key.path)
-      if self._path_filter.CheckPath(normalized_key_path):
-        matching_plugin = self._plugin_per_key_path[normalized_key_path]
 
-      else:
-        for plugin in self._plugins_without_key_paths:
-          if self._CanProcessKeyWithPlugin(registry_key, plugin):
-            matching_plugin = plugin
-            break
+  def _ParseKeysFromFindSpecs(self, parser_mediator, win_registry, find_specs):
+    """Parses the Registry keys from FindSpecs.
 
-      if not matching_plugin:
-        matching_plugin = self._default_plugin
+    Args:
+      parser_mediator (ParserMediator): parser mediator.
+      win_registry (dfwinreg.WinRegistryKey): root Windows Registry key.
+      find_specs (dfwinreg.FindSpecs): Keys to search for.
+    """
+    searcher = dfwinreg_registry_searcher.WinRegistrySearcher(win_registry)
+    for registry_key_path in list(searcher.Find(find_specs=find_specs)):
```

Comment:
could you use iter() instead of list() ?

Reply:
Done.